File: backend/src/main/java/ru/bowling/bowlingapp/DTO/WorkLogSearchDTO.java
package ru.bowling.bowlingapp.DTO;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class WorkLogSearchDTO {

    private Long clubId;
    private Integer laneNumber;
    private Long mechanicId;
    private Long equipmentId;
    private String status;
    private String workType;
    private Integer priority;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private Boolean completedOnly;
    private Boolean activeOnly;
    private String keyword; // для поиска по описанию
    private Boolean includeManualEdits;

    // Параметры пагинации
    @Builder.Default
    private Integer page = 0;
    @Builder.Default
    private Integer size = 20;
    @Builder.Default
    private String sortBy = "createdDate";
    @Builder.Default
    private String sortDirection = "DESC";
}

File: frontend/lib/models/work_log_search_dto.dart
class WorkLogSearchDto {
  int? clubId;
  int? laneNumber;
  int? mechanicId;
  int? equipmentId;
  String? status;
  String? workType;
  int? priority;
  DateTime? startDate;
  DateTime? endDate;
  bool? completedOnly;
  bool? activeOnly;
  String? keyword;
  bool? includeManualEdits;
  int page;
  int size;
  String sortBy;
  String sortDirection;

  WorkLogSearchDto({
    this.clubId,
    this.laneNumber,
    this.mechanicId,
    this.equipmentId,
    this.status,
    this.workType,
    this.priority,
    this.startDate,
    this.endDate,
    this.completedOnly,
    this.activeOnly,
    this.keyword,
    this.includeManualEdits,
    this.page = 0,
    this.size = 20,
    this.sortBy = 'createdDate',
    this.sortDirection = 'DESC',
  });

  Map<String, dynamic> toJson() => {
        'clubId': clubId,
        'laneNumber': laneNumber,
        'mechanicId': mechanicId,
        'equipmentId': equipmentId,
        'status': status,
        'workType': workType,
        'priority': priority,
        'startDate': startDate?.toIso8601String(),
        'endDate': endDate?.toIso8601String(),
        'completedOnly': completedOnly,
        'activeOnly': activeOnly,
        'keyword': keyword,
        'includeManualEdits': includeManualEdits,
        'page': page,
        'size': size,
        'sortBy': sortBy,
        'sortDirection': sortDirection,
      };
}
